src/main/java/duke/command/MarkAsDoneCommand.java

```diff
@@ -0,0 +1,24 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents a command issued by the user to mark a task as done.
+ */
+public class MarkAsDoneCommand extends Command {
+    private final int index; // The index of the task to be marked as done.
```

Comment:
Perhaps `taskIndex` would be more descriptive? For this and other `index` variables in the Command classes.